libraries/ms-helper/inc/can_uart.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+#include "can_hw.h"
+#include "status.h"
+#include "uart.h"
+
+// Generic CAN HW <-> UART protocol
+// Requires CAN HW, UART to be initialized
+
+// Process RX'd CAN message from UART
+struct CanUart;
+typedef void (*CanUartRxCb)(const struct CanUart *can_uart, uint32_t id, bool extended,
+                            const uint64_t *data, size_t dlc, void *context);
+
+typedef struct CanUart {
+  UARTPort uart;
+  CanUartRxCb rx_cb;
+  void *context;
+} CanUart;
+
+// Note that the provided UART port will have its RX handler overwritten.
+StatusCode can_uart_init(CanUart *can_uart);
```

Comment:
@karlding - not sure where your comment went, but the intention is that:

* `can_uart_init`: Initializes module
* `can_uart_hook_can_hw`: Uses an initialized module, takes over the CAN HW implementation

This distinction allows the module to be used for masters and slaves, where `can_uart_hook_can_hw` handles all the CAN HW RX/TX to/from UART. The alternative is to use `CanUartRxCb` to handle received RX messages + TX requests from UART.